### ISSUE TITLE
Accessing element.ref was removed in React 19. ref is now a regular prop. It will be removed from the JSX Element type in a future release

### ISSUE THREAD

**Issue by alekseykurylev** [react-component/menu#744]
Hi guys! Thanks for your component, it's very cool! I would like to know when you will make React 19 support?

Error: "Accessing element.ref was removed in React 19. ref is now a regular prop. It will be removed from the JSX Element type in a future release"

![image](https://github.com/user-attachments/assets/7730b9a8-4cb5-4bcd-9168-982b95755ce6)

**Comment by linhtrinh18** [comment]
Any update on this? I faced the same issue

**Comment by alekseykurylev** [comment]
> Any update on this? I faced the same issue

Yes, the problem remains. In ant-design, it is somehow solved: https://github.com/ant-design/ant-design/blob/master/components/menu/menu.tsx but I'm waiting for the "rc-menu" update
